backends/ebpf/ebpftargets.py

```diff
@@ -207,120 +208,141 @@ def check_outputs(self):
-        self._create_bridge(ifname)
-        # Use clang to compile the generated C code to a LLVM IR
-        result = self._compile_ebpf(self.ebpfdir)
         if result != SUCCESS:
-            self._remove_bridge(ifname)
             return result
-        result = self._load_ebpf(self.ebpfdir, ifname)
+        # Check if eBPF maps have actually been created
+        result = bridge.ns_proc_write(proc,
+                                      "ls -1 /sys/fs/bpf/tc/globals")
         if result != SUCCESS:
-            self._remove_bridge(ifname)
-        return result
+            return result
+        # Finally, append the actual runtime command to the process
+        result = bridge.ns_proc_append(proc, self._get_run_cmd())
+        if result != SUCCESS:
+            return result
+        # Execute the command queue and close the process, retrieve result
+        return bridge.ns_proc_close(proc)
 
-    def check_outputs(self):
-        # Not implemented yet, just pass the test
-        ifname = str(os.getpid())
-        self._remove_bridge(ifname)
-        return SUCCESS
+    def run(self):
+        # Root is necessary to load ebpf into the kernel
+        require_root(self.outputs)
+        result = self._create_runtime()
+        if result != SUCCESS:
+            return result
+        # Create the namespace and the central testing bridge
+        bridge = self._create_bridge()
+        if not bridge:
+            return FAILURE
+        # Run the program in the generated namespace
+        result = self._run_in_namespace(bridge)
+        bridge.ns_del()
+        return result
 
 
 class EBPFBCCTarget(EBPFTarget):
     def __init__(self, tmpdir, options, template, outputs):
         EBPFTarget.__init__(self, tmpdir, options, template, outputs)
 
-    def compile_p4(self, argv):
-        # To override
-        """ Compile the p4 target """
-        if not os.path.isfile(self.options.p4filename):
-            raise Exception("No such file " + self.options.p4filename)
-        args = ["./p4c-ebpf"]
-        args.extend(["--target", self.options.target])
-        args.extend(["-o", self.template])
-        args.append(self.options.p4filename)
-        args.extend(argv)
-        result = run_timeout(self.options, args, TIMEOUT,
-                             self.outputs, "Failed to compile P4:")
-        if result != SUCCESS:
-            print("".join(open(self.outputs["stderr"]).readlines()))
-            # If the compiler crashed fail the test
-            if 'Compiler Bug' in open(self.outputs["stderr"]).readlines():
-                sys.exit(FAILURE)
-        # Check if we expect the p4 compilation of the p4 file to fail
-        expected_error = is_err(self.options.p4filename)
-        if expected_error:
-            # We do, so invert the result
-            if result == SUCCESS:
-                result = FAILURE
-            else:
-                result = SUCCESS
-        return result, expected_error
+    def create_filter(self):
```

Comment:
Are we even trying to run these?
Maybe you can just not implement this class for now.

Reply:
Yeah, both kernel and test use it, the problem is that there is no real abstract method in Python I guess. The run-ebpf-test run_model() function calls all of these operations, which change depending on the target, so implementing it is required.